nomenclature/codelist.py

```diff
@@ -543,8 +543,21 @@ def from_directory(cls, name: str, path: Path, file_glob_pattern: str = "**/*"):
         return cls(name=name, mapping=mapping)
 
     @property
-    def hierarchy(self):
-        raise NotImplementedError("This method is not yet implemented.")
+    def hierarchy(self) -> List[str]:
+        """Return List of strings of available hierarchies to filter
```

Comment:
It's obvious that the returned object is a list of strings from the signature, no need to repeat. Also, the method is not only relevant for filtering, so no to write that
```suggestion
        """Return the hierarchies defined in the RegionCodeList
```